bare-metal/system_timer.c

```diff
@@ -35,19 +40,59 @@ typedef struct sys_timer_struct_s {
     ns_list_link_t link;
 } sys_timer_struct_s;
 
+#define TIMER_SLOTS_PER_MS			20
 #define TIMER_SYS_TICK_PERIOD       10 // milliseconds
-#define TIMER_SYS_TICK_SLOTS        (TIMER_SYS_TICK_PERIOD * 20) // 50us slots
+#define TIMER_SYS_TICK_SLOTS        (TIMER_SYS_TICK_PERIOD * TIMER_SLOTS_PER_SECOND) // 50us slots
 
 static uint32_t run_time_tick_ticks = 0;
 static NS_LIST_DEFINE(system_timer_free, sys_timer_struct_s, link);
 static NS_LIST_DEFINE(system_timer_list, sys_timer_struct_s, link);
-static int8_t sys_timer_id = -1;
 
 
 static sys_timer_struct_s *sys_timer_dynamically_allocate(void);
-static void timer_sys_interrupt(int8_t timer_id, uint16_t slots);
+static void timer_sys_interrupt(void);
+
+#ifndef EVENTLOOP_USE_TICK_TIMER
+/* Implement platform tick timer using eventOS timer */
+int8_t platform_tick_timer_start(uint32_t milliseconds);
+// platform tick timer callback function
+static void (*tick_timer_callback)(void);
+static int8_t tick_timer_id = -1;	// eventOS timer id for tick timer
+
+// EventOS timer callback function
+static void tick_timer_eventOS_callback(int8_t timer_id, uint16_t slots)
+{
+    // Not interested in timer id or slots
+    (void)slots;
+    // Call the tick timer callback
+    if (NULL != tick_timer_callback && timer_id == tick_timer_id) {
+        platform_tick_timer_start(TIMER_SYS_TICK_PERIOD);
```

Comment:
I was debating this a bit as well and didn't notice that this might cause issue with existing platforms. Not really feeling like introducing any potential regression but I guess we can do this if we keep an eye out for possible issues...